src/loader/mod.rs

```diff
@@ -306,6 +306,11 @@ impl KernelLoader for Elf {
     }
 }
 
+#[cfg(feature = "elf")]
+#[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
+// Size of string "PVHNote", including the terminating NULL.
+const PVH_NOTE_STR_SZ: usize = 8;
+
```

Comment:
I agree that checking for both name and type when searching for the PVH note is the right approach, for the reason @jiangliu mentions in the commit message. This will avoid any future collisions due to another namespace using type 0x12 as well.

This patch will not work correctly yet, because the name specified in the [PVH ABI](https://xenbits.xen.org/docs/unstable/misc/pvh.html) is actually just "Xen". So the code above must be changed to:
```rust
// Size of string "Xen", including the terminating NULL.
const PVH_NOTE_STR_SZ: usize = 4;
```
The unit tests I added on [this commit](https://github.com/rust-vmm/linux-loader/commit/0ce5bfa70d1073a9743fadfea233edef09c81e49) use "PVHNote" as the name to generate the test_*note.bin binaries and they must be corrected. I can generate new binaries and share them, or they can be built using the instructions in the commit. Please let me know if you'd like me to do this.

Reply:
@jiangliu I built new binaries that are needed to fix the unit tests. Binaries, source and instructions on how to build are here:
https://gist.github.com/aljimenezb/9918b8562070cb89ab6696025305e3f9

Reply:
Yes, the linux kernel use "Xen" and I used "Xen" too at first, but it fails the unit tests:(
I will update it in next version.

Reply:
@aljimenezb can you add the instructions about generating the binaries to this repository as well? It would be easier if someone else needs to generate some in the future.

Reply:
@andreeaflorescu Sure, I'll open a separate PR for that change. That content feels a little bit out of place in the main README, but we can figure out the details on the discussion.

Reply:
Yeah, maybe we can have a contributing document? Or development?

Another idea would be for those binaries to be generated at build time. I am not a big fun of having binaries in repositories.

Reply:
> Another idea would be for those binaries to be generated at build time. I am not a big fun of having binaries in repositories.

That was my first instinct, but I saw test_elf.bin was already in use I followed that simpler approach instead.

 I am not familiar with the capabilities of inline assembly in Rust, so I'd like feedback from others here as to whether or not it is possible to build the [assembly source code](https://gist.github.com/aljimenezb/9918b8562070cb89ab6696025305e3f9) using Rust inline assembly templates.